tests/test_transform.py

```diff
@@ -0,0 +1,250 @@
+        (1.0, LogUniformDistribution(1, 10)),
+        (0.2, DiscreteUniformDistribution(0, 1, q=0.2)),
+    ],
+)
+def test_search_space_transform_numerical(
+    transform_log: bool,
+    transform_step: bool,
+    param: Any,
+    distribution: BaseDistribution,
+) -> None:
+    trans = _SearchSpaceTransform({"x0": distribution}, transform_log, transform_step)
+
+    expected_low = distribution.low  # type: ignore
+    expected_high = distribution.high  # type: ignore
+
+    if isinstance(distribution, LogUniformDistribution):
+        if transform_log:
+            expected_low = math.log(expected_low)
+            expected_high = math.log(expected_high)
+    elif isinstance(distribution, DiscreteUniformDistribution):
+        if transform_step:
+            half_step = 0.5 * distribution.q
+            expected_low -= half_step
+            expected_high += half_step
+    elif isinstance(distribution, IntUniformDistribution):
+        if transform_step:
+            half_step = 0.5 * distribution.step
+            expected_low -= half_step
+            expected_high += half_step
+    elif isinstance(distribution, IntLogUniformDistribution):
+        if transform_step:
+            half_step = 0.5
+            expected_low -= half_step
+            expected_high += half_step
+        if transform_log:
+            expected_low = math.log(expected_low)
+            expected_high = math.log(expected_high)
+
+    for bound in trans.bounds:
+        assert bound[0] == expected_low
+        assert bound[1] == expected_high
+
+    trans_params = trans.transform({"x0": param})
+    assert trans_params.size == 1
+
+    if isinstance(distribution, (IntUniformDistribution, IntLogUniformDistribution)):
+        assert expected_low <= trans_params <= expected_high
+    else:
+        # TODO(hvy): Change second `<=` to `<` when `suggest_float` is fixed.
+        assert expected_low <= trans_params <= expected_high
+
+
+@pytest.mark.parametrize("transform_log", [True, False])
+@pytest.mark.parametrize("transform_step", [True, False])
+@pytest.mark.parametrize(
+    "param,distribution",
+    [
+        ("foo", CategoricalDistribution(["foo"])),
+        ("bar", CategoricalDistribution(["foo", "bar", "baz"])),
+    ],
+)
+def test_search_space_transform_values_categorical(
+    transform_log: bool,
+    transform_step: bool,
+    param: Any,
+    distribution: CategoricalDistribution,
+) -> None:
+    trans = _SearchSpaceTransform({"x0": distribution}, transform_log, transform_step)
+
+    for bound in trans.bounds:
+        assert bound[0] == 0.0
+        assert bound[1] == 1.0
+
+    trans_params = trans.transform({"x0": param})
+
+    for trans_param in trans_params:
+        assert trans_param in (0.0, 1.0)
+
+
+@pytest.mark.parametrize("transform_log", [True, False])
+@pytest.mark.parametrize("transform_step", [True, False])
+def test_search_space_transform_shapes_dtypes_values_categorical_with_other_distribution(
```

Comment:
It is not easy to understand what this test method checks. I found code smells of "Conditional Test Logic" anti-pattern which is introduced by xUnit test patterns.
http://xunitpatterns.com/Conditional%20Test%20Logic.html

Could you make this test case more simple?

Reply:
Sure, we could for instance unroll the loops and perhaps skip the parameterization. Or actually, we could remove this test altogether. In the initial implementation using scikit-learn, categorical one-hot encoding were always placed in the first rows of the bounds ndarray. I wanted to check that the bounds were still correct.

Reply:
Removed the entire function.